msu/systems/system.nut

```diff
@@ -0,0 +1,40 @@
+::MSU.Class.System <- class
+{
+	ID = null;
+	constructor( _id, _dependencies = null )
+	{
+		if (_dependencies == null) _dependencies = []
+
+		foreach (dependency in _dependencies)
+		{
+			local found = false;
+			foreach (system in ::MSU.System)
+			{
+				if (dependency == system.getID())
+				{
+					found = true;
+					break;
+				}
+			}
+			if (!found)
+			{
+				throw "Dependencies not yet initialized";
+			}
+		}
+		this.ID = _id;
+	}
+
+	function getID()
+	{
+		return this.ID;
+	}
+
+	function registerMod( _mod )
+	{
+		if (!(_mod.getID() in ::MSU.Mods))
```

Comment:
this should never happen now that mods register upon initialization. Maybe we should check if `_mod` is a `::MSU.Class.Mod` instead

Reply:
Yeah makes sense